File: julia/problem_10.jl
# Copyright 2014 Mitchell Kember. Subject to the MIT License.
# Project Euler: Problem 10
# Summation of primes

module problem_10

import problem_07: isprime

function solve()
	const max = 2_000_000
	total = 2 + 3
	n = 1
	while true
		for x in [4, 2]
			n += x
			if n >= max
				return total
			end
			if isprime(n)
				total += n
			end
		end
	end
end

end # module

File: rust/problem_10.rs
// Copyright 2014 Mitchell Kember. Subject to the MIT License.
// Project Euler: Problem 10
// Summation of primes

use problem_07::is_prime;

pub fn solve() -> int {
	let max = 2000000;
	let mut sum = 2 + 3;
	let mut n = 1;
	loop {
		n += 4;
		if n >= max { break; }
		if is_prime(n) { sum += n; }
		n += 2;
		if n >= max { break; }
		if is_prime(n) { sum += n; }
	}
	sum
}
